community/bolt/src/main/java/org/neo4j/bolt/BoltKernelExtension.java

```diff
@@ -177,10 +177,14 @@ public Lifecycle newInstance( KernelContext context, Dependencies dependencies )
                 } )
                 .collect( toList() );
 
-        if ( connectors.size() > 0 )
+        if ( connectors.size() > 0 && !config.get( GraphDatabaseSettings.disconnected ) )
```

Comment:
@nigelsmall should we really do all that work before here if the setting is going to tell us that it's disabled?

What I'm thinking is we could check for `config.get( GraphDatabaseSettings.disconnected )` on line 130 or 137 instead and return early from the method.

But this will work (since the effective line is 182).